tools/assetcompressor/main.cpp

```diff
@@ -0,0 +1,157 @@
+            QObject::tr(
+                    "\nThis tool re-exports glTF 2.0 files with meshes compressed with the Draco algorithm.\n"
+                    "Example : ./assetcompressor -i my_file.gltf -o /path/to/new/folder -q 8"));
+
+    QCommandLineOption inputOption({ "i", "input" }, QObject::tr("glTF2 <file> to compress"), QObject::tr("file"), QString());
+    QCommandLineOption outputOption({ "o", "output" }, QObject::tr("Output <directory>"), QObject::tr("directory"), QString());
+    QCommandLineOption ratioOption({ "q", "quantization" }, QObject::tr("Compression quantization, between 1 (more compressed) and 16 (less compressed)"), QObject::tr("value"), "7");
+
+    cmdline.addOptions({ inputOption, outputOption, ratioOption });
+
+    cmdline.process(app);
+
+    const auto asset = cmdline.value(inputOption);
+    const QFileInfo assetFileInfo(asset);
+    if (!cmdline.isSet(inputOption) || asset.isEmpty() || !assetFileInfo.exists()) {
+        qCritical("Invalid input file.\n%s", cmdline.helpText().toLatin1().constData());
+        return 1;
+    }
+
+    const auto outputDirArg = cmdline.value(outputOption);
+    if (!cmdline.isSet(outputOption) || outputDirArg.isEmpty()) {
+        qCritical("Invalid output directory.\n%s", cmdline.helpText().toLatin1().constData());
+        return 1;
+    }
+
+    const int ratio = cmdline.value(ratioOption).toInt();
+    if (ratio < 1 || ratio > 16) {
+        qCritical("Invalid compression ratio.\n%s", cmdline.helpText().toLatin1().constData());
+        return 1;
+    }
+
+    const QDir targetDir = outputDirArg;
+    if (!targetDir.exists()) {
+        if (!QDir{}.mkpath(outputDirArg)) {
+            qCritical("Could not create output directory.");
+            return 1;
+        }
+    }
+
+    // Parsing
+    SceneEntity scene;
+    GLTF2Context ctx;
+
+    GLTF2Parser parser(&scene);
+    parser.setContext(&ctx);
+
+    const auto res = parser.parse(asset);
+    if (!res) {
+        qCritical("Could not parse glTF2 file.");
+        return 1;
+    }
+
+    // Compression
+    GLTF2ExportConfiguration configuration;
+    configuration.setMeshCompressionEnabled(true);
+    configuration.setAttributeQuantizationLevel(GLTF2ExportConfiguration::Position, ratio);
+    configuration.setAttributeQuantizationLevel(GLTF2ExportConfiguration::Normal, ratio);
+    configuration.setAttributeQuantizationLevel(GLTF2ExportConfiguration::Color, ratio);
+    configuration.setAttributeQuantizationLevel(GLTF2ExportConfiguration::TextureCoordinate, ratio);
+    configuration.setAttributeQuantizationLevel(GLTF2ExportConfiguration::Generic, ratio);
+
+    GLTF2Exporter exporter;
+    exporter.setContext(&ctx);
+    exporter.setScene(&scene);
+    exporter.setConfiguration(configuration);
+
+    const auto sourceDir = assetFileInfo.dir();
+
+    // Meaasure the actual export step duration
+    QElapsedTimer timer;
+    timer.start();
+
+    const auto new_asset = exporter.saveInFolder(sourceDir, targetDir);
```

Comment:
should this fail is any of the output files exists unless a `-f|--force` option is added?

Reply:
that would be safer indeed, I will have to finish https://github.com/KDAB/kuesa/pull/49 first

Reply:
done